keystonemiddleware/auth_token/__init__.py

```diff
@@ -772,6 +777,14 @@ def _get_user_token_from_header(self, env):
         else:
             raise exc.InvalidToken(_('Unable to find token in headers'))
 
+    def _get_client_ip(self, env):
+        ips = self._get_header(env, 'X-Forwarded-For')
+        if ips:
+            return ips.split(',')[0]
```

Comment:
if there is no 'X-forwarded-For' header in request then we should not include NULL IP in the header . We can just avoid filling this header in this case